src/main/java/dev/j3fftw/headlimiter/Utils.java

```diff
@@ -100,7 +102,24 @@ public static void onCheck(Player player, Block block, int maxAmount, int count,
             });
 
             BlockStorage.clearBlockInfo(block.getLocation());
-            player.sendMessage(ChatColor.RED + "You hit the limit of Cargo nodes in this chunk");
+            if (isPlacingRestricted) {
+                player.sendMessage(ChatColor.RED + "You can't place Cargo nodes in unprotected areas!");
+            } else {
+                player.sendMessage(ChatColor.RED + "You hit the limit of Cargo nodes in this chunk");
+            }
         }
     }
+
+    /**
+     * Whether the block placement is prohibited or not by protection plugins
+     * @param block The block to be checked
+     * @return Whether the placement is prohibited or not
+     */
+    public static boolean isPlacingRestricted(@Nonnull Block block) {
+        // This is intentionally redundant to allow for expandability by adding more booleans and returning their || chain
```

Comment:
```suggestion
        // This is intentionally redundant to allow for expand ability by adding more booleans and returning their || chain
```
I found this when i was typing the other comment apparently its typed like that